Compute features option not working in Mastodon

Hi Jean Yves,
I have been using Mastodon plugin in FIJI to track cells manually in zebrafish embryos. When I finished tracking and went for “Compute Features” option, I get the following error:

Exception in thread “Feature computation thread” java.lang.NoClassDefFoundError: io/humble/ferry/AtomicInteger
at org.mastodon.mamut.feature.SpotTrackIDFeatureComputer.run(SpotTrackIDFeatureComputer.java:77)
at org.mastodon.feature.DefaultFeatureComputerService.compute(DefaultFeatureComputerService.java:261)
at org.mastodon.mamut.feature.MamutFeatureComputerService.compute(MamutFeatureComputerService.java:96)
at org.mastodon.mamut.MamutFeatureComputation$MyFeatureComputerService.compute(MamutFeatureComputation.java:163)
at org.mastodon.feature.ui.FeatureComputationController$2.run(FeatureComputationController.java:129)
Caused by: java.lang.ClassNotFoundException: io.humble.ferry.AtomicInteger
at java.net.URLClassLoader.findClass(URLClassLoader.java:387)
at java.lang.ClassLoader.loadClass(ClassLoader.java:419)
at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:352)
at java.lang.ClassLoader.loadClass(ClassLoader.java:352)
… 5 more

I have updated java, updated fiji, tried uninstalling and re-installing Mastodon, but still have the same problem. Any suggestions?

Thanks
Sundar

--- a/src/main/java/org/mastodon/mamut/feature/SpotTrackIDFeatureComputer.java
+++ b/src/main/java/org/mastodon/mamut/feature/SpotTrackIDFeatureComputer.java
@@ -43,7 +43,7 @@ import org.scijava.ItemIO;
 import org.scijava.plugin.Parameter;
 import org.scijava.plugin.Plugin;
 
-import io.humble.ferry.AtomicInteger;
+import java.util.concurrent.atomic.AtomicInteger;
 
 @Plugin( type = MamutFeatureComputer.class )
 public class SpotTrackIDFeatureComputer implements MamutFeatureComputer

--- a/src/main/java/org/mastodon/mamut/feature/branch/BranchDisplacementDurationFeatureComputer.java
+++ b/src/main/java/org/mastodon/mamut/feature/branch/BranchDisplacementDurationFeatureComputer.java
@@ -28,6 +28,7 @@
  */
 package org.mastodon.mamut.feature.branch;
 
+import net.imglib2.util.Util;
 import org.mastodon.mamut.feature.MamutFeatureComputer;
 import org.mastodon.mamut.model.Model;
 import org.mastodon.mamut.model.ModelGraph;
@@ -94,19 +95,8 @@ public class BranchDisplacementDurationFeatureComputer implements MamutFeatureCo
 		// get target spot
 		final Spot target = branchGraph.getLastLinkedVertex( branchSpot, ref2 );
 
-		output.dispMap.set( branchSpot, distance( source, target ) );
+		output.dispMap.set( branchSpot, Util.distance( source, target ) );
 		output.durMap.set( branchSpot, duration( source, target ) );
-	}
-
-	private double distance( Spot source, Spot target )
-	{
-		double d2 = 0.;
-		for ( int d = 0; d < 3; d++ )
-		{
-			final double dx = source.getDoublePosition( d ) - target.getDoublePosition( d );
-			d2 += dx * dx;
-		}
-		return Math.sqrt( d2 );
 	}
 
 	private double duration( Spot source, Spot target )

--- a/src/main/java/org/mastodon/mamut/views/bdv/MamutBranchViewBdv.java
+++ b/src/main/java/org/mastodon/mamut/views/bdv/MamutBranchViewBdv.java
@@ -49,6 +49,7 @@ import org.mastodon.graph.GraphIdBimap;
 import org.mastodon.mamut.MainWindow;
 import org.mastodon.mamut.MamutMenuBuilder;
 import org.mastodon.mamut.ProjectModel;
+import org.mastodon.mamut.TimepointAndNumberOfSpotsPanel;
 import org.mastodon.mamut.model.BoundingSphereRadiusStatistics;
 import org.mastodon.mamut.model.BranchGraphModelOverlayProperties;
 import org.mastodon.mamut.model.Link;
@@ -209,6 +210,11 @@ public class MamutBranchViewBdv extends MamutBranchView<
 		BdvSelectionBehaviours.install( viewBehaviours, viewGraph, tracksOverlay, selectionModel, focusModel,
 				navigationHandler );
 		OverlayActions.install( viewActions, viewer, tracksOverlay );
+
+		// Add the timepoint and number of spots panel.
+		final TimepointAndNumberOfSpotsPanel timepointAndNumberOfSpotsPanel =
+				new TimepointAndNumberOfSpotsPanel( this.timepointModel, model );
+		frame.getSettingsPanel().add( timepointAndNumberOfSpotsPanel );
 
 		/*
 		 * We must make a search action using the underlying model graph,

--- a/src/main/java/org/mastodon/mamut/views/bdv/MamutViewBdv.java
+++ b/src/main/java/org/mastodon/mamut/views/bdv/MamutViewBdv.java
@@ -48,6 +48,7 @@ import org.mastodon.app.ui.ViewMenuBuilder.JMenuHandle;
 import org.mastodon.mamut.MainWindow;
 import org.mastodon.mamut.MamutMenuBuilder;
 import org.mastodon.mamut.ProjectModel;
+import org.mastodon.mamut.TimepointAndNumberOfSpotsPanel;
 import org.mastodon.mamut.UndoActions;
 import org.mastodon.mamut.model.Link;
 import org.mastodon.mamut.model.Model;
@@ -231,6 +232,11 @@ public class MamutViewBdv
 		final Runnable onCloseMIPDialog = RecordMaxProjectionMovieDialog.install( viewActions, bdv, tracksOverlay,
 				colorBarOverlay, appModel.getKeymap() );
 		onClose( onCloseMIPDialog );
+
+		// Add the timepoint and number of spots panel.
+		final TimepointAndNumberOfSpotsPanel timepointAndNumberOfSpotsPanel =
+				new TimepointAndNumberOfSpotsPanel( this.timepointModel, model );
+		frame.getSettingsPanel().add( timepointAndNumberOfSpotsPanel );
 
 		/*
 		 * We must make a search action using the underlying model graph,

